app/domain/authentication/authn_k8s/extract_k8s_resource_restrictions.rb

```diff
@@ -0,0 +1,70 @@
+require 'command_class'
+
+module Authentication
+  module AuthnK8s
+
+    ExtractK8sResourceRestrictions = CommandClass.new(
+      dependencies: {
+        extract_resource_restrictions: Authentication::ResourceRestrictions::ExtractResourceRestrictions.new,
+        resource_restrictions_class:   Authentication::ResourceRestrictions::ResourceRestrictions,
+        logger:                        Rails.logger
+      },
+      inputs:   %i(authenticator_name service_id role_name account)
+    ) do
+
+      def call
+        extract_resource_restrictions
+      end
+
+      private
+
+      # Extracts resource restrictions for k8s hosts. It can be defined in the
+      # annotations or as part of the host ID itself.
+      # If the restrictions are in the annotations, then they will not be
+      # extracted from the host ID.
+      def extract_resource_restrictions
+        resource_restrictions = @extract_resource_restrictions.call(
```

Comment:
i would add a comment saying that this extracts the resource restrictions from the annotations

Reply:
done

Reply:
I propose we call this a "Pinocchio" comment.

Just like Pinocchio was a puppet who wanted to be real boy, sometimes comments want to be a real variable name.

Perhaps `get_restrictions_from_annotations`?

Reply:
done